server/src/main/java/io/druid/client/HttpServerInventoryView.java

```diff
@@ -564,12 +514,42 @@ public void onSuccess(InputStream stream)
                     }
 
                     if (counter == null) {
-                      druidServer.removeAllSegments();
-                    }
+                      Map<String, DataSegment> toRemove = Maps.newHashMap(druidServer.getSegments());
```

Comment:
Would you add a comment what null counter means?

Reply:
Also, it would be great if you split the logic in ```onSuccess()``` into some smaller methods because it will be easier to understand what's going on here.

Reply:
adding a comment.